mio/process/video.py

```diff
@@ -0,0 +1,429 @@
+
+            subtract_minimum = VideoProcessor.normalize_video_stack(subtract_minimum)
+
+            raw_video = NamedFrame(name="RAW", video_frame=raw_frames)
+
+            if config.noise_patch.enable:
+                patched_video = NamedFrame(name="patched", video_frame=patched_frames)
+                if config.noise_patch.output_result:
+                    patched_video.export(
+                        output_dir / f"{pathstem}",
+                        suffix=True,
+                        fps=20,
+                    )
+            if config.noise_patch.output_diff:
+                diff_video = NamedFrame(
+                    name=f"diff_{config.noise_patch.diff_multiply}x", video_frame=diff_frames
+                )
+                diff_video.export(
+                    output_dir / f"{pathstem}",
+                    suffix=True,
+                    fps=20,
+                )
+            if config.noise_patch.output_noise_patch:
+                noise_patch = NamedFrame(name="noise_patch", video_frame=noise_patchs)
+            if config.frequency_masking.output_mask:
+                freq_mask_frame = NamedFrame(
+                    name="freq_mask", static_frame=freq_mask * np.iinfo(np.uint8).max
+                )
+                freq_mask_frame.export(
+                    output_dir / f"{pathstem}",
+                    suffix=True,
+                    fps=20,
+                )
+
+            if config.frequency_masking.enable:
+                freq_domain_video = NamedFrame(name="freq_domain", video_frame=freq_domain_frames)
+                freq_filtered_video = NamedFrame(
+                    name="freq_filtered", video_frame=freq_filtered_frames
+                )
+                if config.frequency_masking.output_freq_domain:
+                    freq_domain_video.export(
+                        output_dir / f"{pathstem}",
+                        suffix=True,
+                        fps=20,
+                    )
+                if config.frequency_masking.output_result:
+                    freq_filtered_video.export(
+                        (output_dir / f"{pathstem}"),
+                        suffix=True,
+                        fps=20,
+                    )
+
+            min_proj_frame = NamedFrame(name="min_proj", static_frame=minimum_projection)
+
+            if config.interactive_display.enable:
+                videos = [
+                    raw_video,
+                    noise_patch,
+                    patched_video,
+                    freq_filtered_video,
+                    freq_domain_video,
+                    min_proj_frame,
+                    freq_mask_frame,
+                ]
+                VideoPlotter.show_video_with_controls(
+                    videos,
+                    start_frame=config.interactive_display.start_frame,
+                    end_frame=config.interactive_display.end_frame,
+                )
+
```

Comment:
here's a good example of why we want all these things to be separable functions/classes/etc. that can work framewise - ideally this would work on demand, where we just open the UI and compute it as requested in the context of a GUI (not something `mio` should try and provide) rather than needing to precompute everything and set up a UI for it afterwards.

Reply:
Yeah makes sense. This will be mainly for debugging, as the interactive plot gets very slow after a few hundred frames. I guess I'll put non-user facing stuff in the debug section of the config.